### ISSUE TITLE
Symfony 3 Could not load type "sonata_page_create_snapshot" when creating new publication of a page

### ISSUE THREAD

**Issue by vdidry** [sonata-project/SonataPageBundle#848]
#### Sonata packages

"sonata-project/admin-bundle": "^3.20",
"sonata-project/cache-bundle": "^2.3",
"sonata-project/core-bundle": "^3.4",
"sonata-project/datagrid-bundle": "2.2.*@dev",
"sonata-project/doctrine-orm-admin-bundle": "^3.1",
"sonata-project/easy-extends-bundle": "^2.2",
"sonata-project/notification-bundle": "^3.1",
"sonata-project/page-bundle": "^3.5",
"sonata-project/seo-bundle": "^2.2",

#### Symfony packages

"symfony/monolog-bundle": "^3.1.0",
"symfony/polyfill-apcu": "^1.0",
"symfony/swiftmailer-bundle": "^2.3.10",
"symfony/symfony": "3.3.*",
"twig/twig": "^1.0||^2.0"

#### PHP version

7.1

## Subject

Error 500 : Could not load type "sonata_page_create_snapshot" when creating new publication of page

## Steps to reproduce

1) Create a new page.
2)Go to Publications
3)Add new

## Expected results

New publication of page created

## Actual results

```
Symfony\Component\Form\Exception\InvalidArgumentException:
Could not load type "sonata_page_create_snapshot"

  at vendor/symfony/symfony/src/Symfony/Component/Form/FormRegistry.php:87
  at Symfony\Component\Form\FormRegistry->getType('sonata_page_create_snapshot')
     (vendor/symfony/symfony/src/Symfony/Component/Form/FormFactory.php:58)
  at Symfony\Component\Form\FormFactory->createBuilder('sonata_page_create_snapshot', object(Snapshot), array())
     (vendor/symfony/symfony/src/Symfony/Component/Form/FormFactory.php:30)
  at Symfony\Component\Form\FormFactory->create('sonata_page_create_snapshot', object(Snapshot), array())
     (vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php:331)
  at Symfony\Bundle\FrameworkBundle\Controller\Controller->createForm('sonata_page_create_snapshot', object(Snapshot))
     (vendor/sonata-project/page-bundle/Controller/SnapshotAdminController.php:49)
  at Sonata\PageBundle\Controller\SnapshotAdminController->createAction(object(Request))
  at call_user_func_array(array(object(SnapshotAdminController), 'createAction'), array(object(Request)))
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:153)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:68)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:171)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (web/app_dev.php:29)
```

**Comment by greg0ire** [comment]
Could you make a PR to conditionally load this type by form class instead of by name?

**Comment by acrobat** [comment]
Fixed in #875, can be closed.
